stock_picking_auto_create_lot/models/stock_move.py

```diff
@@ -22,3 +22,22 @@ def _compute_display_assign_serial(self):
         for move in moves_not_display:
             move.display_assign_serial = False
         return
+
+    def _set_quantities_to_reservation(self):
+        """Allow complete quantities reservation if the lot will be created when
+        the picking has been validated
+        """
+        res = super()._set_quantities_to_reservation()
+        moves_to_process = self.filtered(
+            lambda mv: not mv.quantity_done
+            and mv.picking_type_id.use_create_lots
+            and mv.picking_type_id.auto_create_lot
+            and mv.product_id.auto_create_lot
+        )
+        for move in moves_to_process:
+            if move.state not in ("partially_available", "assigned"):
```

Comment:
I think this condition can be moved inside `filtered()` or the content of `filtered()` can be moved inside this for loop.